sycl/source/detail/platform_impl.cpp

```diff
@@ -303,9 +304,20 @@ static void filterDeviceFilter(vector_class<RT::PiDevice> &PiDevices,
   if (!FilterList)
     return;
 
+  // remember the last backend that has gone through tis filter function
+  // to assign a unique device id number across platforms that belong to
+  // the same backend. For example, opencl:cpu:0, opencl:acc:1, opencl:gpu:2
+  static backend lastBackend = backend::all;
   backend Backend = Plugin.getBackend();
   int InsertIDx = 0;
-  int DeviceNum = 0;
+  // DeviceNums should be given consecutive numbers across platforms.
```

Comment:
SYCL_DEVICE_ALLOWLIST affects device list before filterDeviceFilter is called. So device ID will change if someone change  SYCL_DEVICE_ALLOWLIST value. 
(see https://github.com/intel/llvm/pull/3611/files#diff-8000a8ba3352f2f7a7ed9fd42213023debfaa6d3ceafe1c147514d7c68a245eaR409)
It can be a problem? We should should definitely document it.

Reply:
Good point.
I will document this point in EnvVar.md.

Reply:
Done. Thanks.